### PR TITLE
feat(core): Add initial support for realms

### PR DESCRIPTION
**Pull request by andreubotella**
This handles bindings, extension JS and sync ops. It also adds the `JsRealm` API that adds methods like `JsRuntime`'s `handle_scope`, `global_object` and `execute_script` specific to the realm.

Towards #13239.

-------------------

@bartlomieju @AaronO @littledivy PTAL. Does this conflict with some of the remaining work to do for ops?

### CODE REVIEW

**Review by lucacasonato — APPROVED** [review]
Looks good to me. Thanks for the work @andreubotella.

I do wonder if `JsRealm`'s lifetime could not be bound to `JsRuntime`, so it could store a reference to the `JsRuntime` in the `JsRealm` struct. That would alleviate the need to pass the runtime to all the methods on the `JsRealm`.

File: core/runtime.rs [hunk 1]
```diff
@@ -416,9 +421,38 @@ impl JsRuntime {
     self.inspector.as_mut().unwrap()
   }
 
+  pub fn global_realm(&mut self) -> JsRealm {
+    JsRealm::new(self.global_context())
+  }
```
Comment (bartlomieju) [inline core/runtime.rs:426]: Why is it not stored on `JsRuntime`? Seems strange to call into `JsRealm::new` every time you want to access it

Comment (andreubotella) [reply core/runtime.rs]: Done

**Comment by bartlomieju** [comment]
Yikes, sorry it takes so long @andreubotella, could you rebase the PR?

**Comment by andreubotella** [comment]
> Yikes, sorry it takes so long @andreubotella, could you rebase the PR?

Done.

**Review by littledivy — APPROVED** [review]
LGTM!

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**